## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#8013d043-9203-4025-a9f6-72e10f1964f6) |
| mint-large-bucket.sh | :x: [more...](#6e697bf5-3b31-4799-81bf-fb27811fb7b8) |
| mint-compression-fs.sh | :x: [more...](#c212ad75-303b-484c-b048-7755d2435bb5) |
| mint-dist-xl.sh | :x: [more...](#7ce38b81-6e31-4323-b5a4-07f0a4b60e18) |
<a name="8013d043-9203-4025-a9f6-72e10f1964f6"></a>
### 8508-358e181/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:31766
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4667d8ab4b92:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-arf1gepgbokueu",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364271,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D64401D43A7A61, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="7ce38b81-6e31-4323-b5a4-07f0a4b60e18"></a>
### 8508-358e181/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:31853
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 2d5238723834:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 6 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 13 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 1 minutes and 58 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 43 seconds
(8/14) Running minio-dotnet tests ... done in 42 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 35 seconds
(10/14) Running minio-java tests ... done in 2 minutes and 34 seconds
(11/14) Running minio-js tests ... done in 57 seconds
(12/14) Running minio-py tests ... FAILED in 3 minutes and 15 seconds
{
  "name": "minio-py:test_presigned_get_object_response_headers",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1400, in test_presigned_get_object_response_headers\n    returned_content_language = response.headers['Content-Language']\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/_collections.py\", line 156, in __getitem__\n    val = self._container[key.lower()]\nKeyError: 'content-language'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2043, in main\n    test_presigned_get_object_response_headers(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1418, in test_presigned_get_object_response_headers\n    raise Exception(err)\nException: 'content-language'\n",
  "args": {
    "object_name": "a0dfa2ff-dcb8-468e-ab0c-0d0bf0a29859",
    "bucket_name": "minio-py-test-82333642-18bc-4fec-9ef8-8592436b229e",
    "presigned_get_object_url": "http://72.28.97.51:31853/minio-py-test-82333642-18bc-4fec-9ef8-8592436b229e/a0dfa2ff-dcb8-468e-ab0c-0d0bf0a29859?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20191112%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20191112T005824Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&response-content-language=en_US&response-content-type=text%2Fplain&X-Amz-Signature=897539615def0f35cfae58afc0dcd4abf3b52790ff0c7f2a44266c4da9b50ea5"
  },
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "status": "FAIL",
  "message": "'content-language'",
  "duration": 111
}

Executed 11 out of 14 tests successfully.
```
<a name="c212ad75-303b-484c-b048-7755d2435bb5"></a>
### 8508-358e181/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:30990
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 98be9e23bc65:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 5 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 57 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 58 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 26 seconds
(8/14) Running minio-dotnet tests ... done in 25 seconds
(9/14) Running minio-go tests ... done in 36 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 13 seconds
(11/14) Running minio-js tests ... done in 43 seconds
(12/14) Running minio-py tests ... FAILED in 2 minutes and 55 seconds
{
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1345, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2040, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1359, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "status": "FAIL",
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "duration": 57,
  "args": {
    "response.reason": "Forbidden",
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>4219486b-8c40-4b24-9c5f-c41e8f10e0e1</Key><BucketName>minio-py-test-5a73e279-fd0e-4139-ada5-d12b2bd2e511</BucketName><Resource>/minio-py-test-5a73e279-fd0e-4139-ada5-d12b2bd2e511/4219486b-8c40-4b24-9c5f-c41e8f10e0e1</Resource><RequestId>15D6442FA567832D</RequestId><HostId>6d00a3e5-9e2c-4005-b56e-305464191d31</HostId></Error>",
    "presigned_get_object_url": "http://72.28.97.55:30990/minio-py-test-5a73e279-fd0e-4139-ada5-d12b2bd2e511/4219486b-8c40-4b24-9c5f-c41e8f10e0e1?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20191112%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20191112T005436Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=e6ba4b42f4aa04b2f5a7e539791434a85dc724896dc7d6b0e190c7657ba56d22",
    "response.status": 403,
    "object_name": "4219486b-8c40-4b24-9c5f-c41e8f10e0e1",
    "bucket_name": "minio-py-test-5a73e279-fd0e-4139-ada5-d12b2bd2e511",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-11-12T00-26-10Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15D6442FA567832D\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Tue, 12 Nov 2019 00:54:51 GMT\"]}}"
  },
  "message": "AccessDenied: message: Access Denied",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)"
}

Executed 11 out of 14 tests successfully.
```
<a name="6e697bf5-3b31-4799-81bf-fb27811fb7b8"></a>
### 8508-358e181/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:31279
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8d772eec84d5:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 7 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 21 seconds
(4/14) Running aws-sdk-ruby tests ... done in 17 seconds
(5/14) Running awscli tests ... done in 2 minutes and 23 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 51 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 25 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 43 seconds
{
  "args": {},
  "duration": 75246,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8508-358e181

--- a/cmd/admin-handlers-config-kv.go
+++ b/cmd/admin-handlers-config-kv.go
@@ -183,7 +183,9 @@ func (a adminAPIHandlers) SetConfigKVHandler(w http.ResponseWriter, r *http.Requ
 	}
 
 	// Make sure to write backend is encrypted
-	saveConfig(context.Background(), objectAPI, backendEncryptedFile, backendEncryptedMigrationComplete)
+	if globalConfigEncrypted {
+		saveConfig(context.Background(), objectAPI, backendEncryptedFile, backendEncryptedMigrationComplete)
+	}
 }
 
 // GetConfigKVHandler - GET /minio/admin/v2/get-config-kv?key={key}
@@ -447,7 +449,9 @@ func (a adminAPIHandlers) SetConfigHandler(w http.ResponseWriter, r *http.Reques
 	}
 
 	// Make sure to write backend is encrypted
-	saveConfig(context.Background(), objectAPI, backendEncryptedFile, backendEncryptedMigrationComplete)
+	if globalConfigEncrypted {
+		saveConfig(context.Background(), objectAPI, backendEncryptedFile, backendEncryptedMigrationComplete)
+	}
 
 	// Reply to the client before restarting minio server.
 	writeSuccessResponseHeadersOnly(w)

--- a/cmd/config-encrypted.go
+++ b/cmd/config-encrypted.go
@@ -19,11 +19,12 @@ package cmd
 import (
 	"bytes"
 	"context"
+	"errors"
 	"os"
 	"strings"
+	"unicode/utf8"
 
 	etcd "github.com/coreos/etcd/clientv3"
-	jsoniter "github.com/json-iterator/go"
 	"github.com/minio/minio/cmd/config"
 	"github.com/minio/minio/cmd/logger"
 	"github.com/minio/minio/pkg/auth"
@@ -260,11 +261,8 @@ func migrateIAMConfigsEtcdToEncrypted(client *etcd.Client) error {
 			data = cdata
 		}
 
-		// Attempt to unmarshal JSON content
-		var dummy map[string]interface{}
-		var json = jsoniter.ConfigCompatibleWithStandardLibrary
-		if err = json.Unmarshal(data, &dummy); err != nil {
-			return err
+		if !utf8.Valid(data) {
+			return errors.New("config data not in plain-text form")
 		}
 
 		cencdata, err = madmin.EncryptData(globalActiveCred.String(), data)
@@ -335,11 +333,8 @@ func migrateConfigPrefixToEncrypted(objAPI ObjectLayer, activeCredOld auth.Crede
 				data = cdata
 			}
 
-			// Attempt to unmarshal JSON content
-			var dummy map[string]interface{}
-			var json = jsoniter.ConfigCompatibleWithStandardLibrary
-			if err = json.Unmarshal(data, &dummy); err != nil {
-				return err
+			if !utf8.Valid(data) {
+				return errors.New("config data not in plain-text form")
 			}
 
 			cencdata, err = madmin.EncryptData(globalActiveCred.String(), data)

--- a/cmd/logger/config.go
+++ b/cmd/logger/config.go
@@ -193,7 +193,7 @@ func LookupConfig(scfg config.Config) (Config, error) {
 		if starget != config.Default {
 			authTokenEnv = EnvLoggerHTTPAuditAuthToken + config.Default + starget
 		}
-		cfg.HTTP[starget] = HTTP{
+		cfg.Audit[starget] = HTTP{
 			Enabled:   true,
 			Endpoint:  endpoint,
 			AuthToken: env.Get(authTokenEnv, kv.Get(AuthToken)),

